### DISCUSSION TITLE
Dynamically resize input buffer

### DISCUSSION THREAD

**Discussion by raiguard** [YaLTeR/niri#758] [Ideas]
By default, Wayland has a fixed input buffer size, meaning that if a client freezes for too long, it will get disconnected. This happens on all Wayland compositors and is particularly frustrating when debugging, and players of [the game I work on](https://forums.factorio.com/105527) have also ran into the issue due to the game's single-threaded nature.

Fortunately, the Wayland library now supports resizing this input buffer: https://gitlab.freedesktop.org/wayland/wayland/-/merge_requests/188.

Could niri please implement this so that I can keep my sanity when debugging under Wayland?

Video demonstration of how easy it is to encounter this while debugging:

https://github.com/user-attachments/assets/9124cb80-1b3d-43c7-a98c-d7af40f736ce

**Comment by YaLTeR** [comment]
Right, yeah. The question is whether this is something that Smithay needs to do, or on niri side. I'll ask in the Smithay Matrix room.
